src/java.security.jgss/share/classes/sun/security/krb5/internal/KDCRep.java

```diff
@@ -138,7 +138,7 @@ protected void init(DerValue encoding, int req_type)
                         " req type is " + req_type);
 
                 System.out.println(">>> KDCRep: Message in bytes is =>");
-                byte[] dataBytes = encoding.getDataBytes();
+                byte[] dataBytes = encoding.data().toByteArray();
```

Comment:
As I was reviewing the back-ports of this, I noticed that `toByteArray()` might return `null` (if there are no bytes in encoding). I expect an exception to be thrown by
```java
        der = encoding.getData().getDerValue();
```
on line 159 below, but enabling debugging should replace that with a `NullPointerException` on line 143.

Please address that and update the back-ports similarly.

Reply:
I (partially) take that back.; `toByteArray()` never returns `null` in this version, but it might in earlier versions.

Reply:
The `toByteArray()` will only throw a `NullPointerException` if the buffer is null. But this cannot happen since the constructor of the original `DerValue` would throw an `IOException`.

Reply:
That potential problem only applies to Java 11.